### DISCUSSION TITLE
Best version of snow partitioning to use?

### DISCUSSION THREAD

**Discussion by akkendrick** [PMEAL/porespy#403] [Q&A] [answered]
I am trying to associate a vector of normalized 3D flow velocities from a different simulation engine (Palabos) to a single pore in python (identified by SNOW). Using the dictionary produced by ps.networks.snow is super handy for this (because of the output dictionary that contains the pores, apart from the added boundary faces) but it is not parallel as far as I can see.  On the other hand, ps.filters.snow can be parallel but does not make the handy dictionary of pores. What I want to do is make a dictionary of velocities and add values to it for each pore (so each key in velocities is a pore, and for each key there is a list of flow velocities)

I suppose I should just make a dictionary of pores from the filter code and the output regions...?

Is the SNOW implementation the same across both of these methods? Is making a dictionary of pores from the filter code so that it can be parallel and also matches my simulation of velocity the easiest method to do this? 

I want to leverage the network framework but it is more difficult to go this route I think... I could be missing something obvious though... Thanks in advance for your thoughts. This also could just require more coding on my end which is fine I just want to be as efficient as possible.

**Comment by ma-sadeghi** [comment, marked as answer]
Shouldn't you be assigning velocities to the throats rather than pores?

**Reply by jgostick** [reply]
Indeed, you can assign scalar components like pressure or concentration to pores, but velocity in a pore does not really exist.

**Reply by akkendrick** [reply]
At a first pass, this makes sense. The high flow velocities should be in throats with the values in pores approaching some background constant value (I think... pending alignment of pores with the pressure gradient? I need to think about this)

Same questions though... ps.network.snow makes this handy because of the output dictionary but  ps.filters.snow_parallel is parallel but doesn't produce the handy dictionary for OpenPNM. Between network.snow or filters.snow is there a preferred route in your mind that would be best for matching porous media geometry with a simulated flow field?

**Reply by jgostick** [reply]
Is your image so large that you need to do it in parallel?  The parallel snow is for images that are 1000-cubed and above.  

To answer your question, can you call the ``regions_to_network`` function in the ``ps.networks`` module to get the openpnm network dictionary.  The granular workflow of the ``snow`` extraction is explained in [this example](https://github.com/PMEAL/porespy/blob/dev/examples/networks/snow_advanced.ipynb)

**Reply by akkendrick** [reply]
Awesome, sounds good. I will go forward with the `ps.filters.snow` algorithm. I don't really need parallel I think... the max cube size I anticipate is ~500. 

Thanks, this is really helpful. I remember seeing that in the docs awhile back I just didn't associate it with this problem in particular.

**Comment by jgostick** [comment]
BTW, we are in the process of 'tidying' and 'unifying' the network extraction codes in porespy as part of the transition to V2.  These things all take 10x longer than planned though.  The planned changes will address your comments though.
